kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/client/KafkaConnectClients.java

```diff
@@ -0,0 +1,16 @@
+package com.provectus.kafka.ui.cluster.client;
+
+import com.provectus.kafka.ui.connect.ApiClient;
+import com.provectus.kafka.ui.connect.api.ConnectApi;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public final class KafkaConnectClients {
+
+    private static final Map<String, ConnectApi> CACHE = new ConcurrentHashMap<>();
+
+    public static ConnectApi withBaseUrl(String basePath) {
+        return CACHE.computeIfAbsent(basePath, path -> new ConnectApi(new ApiClient().setBasePath(path)));
```

Comment:
I think it's better to expire items in the cache after some time. You can use existing cache implementation in this case

Reply:
Do we really need expiration here?
The client instances are limited to number of connect clusters we provide during startup time